mapbox_vector_tile/decoder.py

```diff
@@ -109,8 +119,16 @@ def parse_geometry(self, geom, ftype):
                     dx = x
                     dy = y
 
-                    coords.append([x, 4096-y])
-
-        if ftype == POLYGON and len(coords) > 0:
-            coords.append(coords[0])
-        return coords
+                    coords.append([x, self.extents-y])
+
+        if ftype == POINT:
+            return coords
+        elif ftype == LINESTRING or ftype == POLYGON:
+            if parts:
+                if coords:
+                    parts.append(coords)
+                return parts[0] if len(parts) == 1 else parts
```

Comment:
This seems to return all the parts for a `POLYGON` type, but don't we need to calculate the area of each to tell whether it's a single Polygon or a MultiPolygon?